shpec/shpec_shpec.sh

```diff
@@ -35,7 +35,8 @@ describe "shpec"
 
   describe "equality matcher"
     it "handles newlines properly"
-      string_with_newline_char="new\nline"
+      string_with_newline_char="new
```

Comment:
On 04/13/15 04:08, Ryland Herrick wrote:
> In shpec/shpec_shpec.sh <https://github.com/rylnd/shpec/pull/56#discussion_r28211219>:
>
>> @@ -35,7 +35,8 @@ describe "shpec"
>>
>>    describe "equality matcher"
>>      it "handles newlines properly"
>> -      string_with_newline_char="new\nline"
>> +      string_with_newline_char="new
>
> Was this intentionally changed? This test was added specifically for #23 <https://github.com/rylnd/shpec/issues/23>.

I've added a comment to the issue.  POSIX just sees two characters there, not a newline.

Bash and ksh support 'ansi-c' strings with the $'...' string notation.

I would recommend to use initialised string constants using `printf`.

     CHAR_NL=$( printf "\n" )

If we do this once at the top of `shpec()`, we could rewrite the assertion above as:

     string_with_newline_char="new${CHAR_NL}line"

etc., etc.

Cheers,
Henk